### DISCUSSION TITLE
How can I set the color property of a material in Matlab/Octave?

### DISCUSSION THREAD

**Discussion by chr-ger** [thliebig/openEMS-Project#141] [Q&A Usage]
Hi,

I'm using a 6-layer PCB and the default coloring in AppCSXCAD makes it really hard to see anything. I'm currently using AddMaterial() instead of AddMetal() for the conductors, but they seem to be displayed transparently. I would also like to be able to select a uniform substrate color  and metal color, at least for the inner layers.

![stackup](https://github.com/thliebig/openEMS-Project/assets/9386066/6fefa586-19a2-46b9-bb0a-b2ae50a41c53)

I found this in the Python implementation of the CSX: 
https://github.com/thliebig/CSXCAD/blob/master/python/CSXCAD/CSProperties.pyx?#L196

Is there any similiar function/property for the AddMaterial() and AddMetal() functions in ocatve? Or any generic way of setting the color and transparency of an CSX property? 

Greets,
Chris

**Comment by oberstet** [comment]
coincidence, I was just looking for exactly what you want with colors, but in Python, and found your question with the link, thanks;)

not helping I guess, but I can confirm at least it works using Python. I wanted both helixes in below to have the same _color_ even though they are _separate_ conductors. I would expect it should work in Octave as well (but I am not using Octave and don't know).

code

```python
helix_y0 = 0
self._helix1: CSPropMetal = self._csx.AddMetal('helix1')
self._helix1.SetColor('#cdbaab', 255)
self._helix1_waypoints = self._setup_helix(-self._groundplane_width / 2 + self._helix_width / 2 + 2, helix_y0, self._helix1)
self._helix2: CSPropMetal = self._csx.AddMetal('helix2')
self._helix2.SetColor('#cdbaab', 255)
self._helix2_waypoints = self._setup_helix(self._groundplane_width / 2 - self._helix_width / 2 - 2, helix_y0, self._helix2)
```

result

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/e644cf38-8c45-452d-ab08-dfb653012fd2)
